src/core/mantle/evaluator/simple/utils.ts

```diff
@@ -0,0 +1,67 @@
+import { pick } from "lodash-es";
+
+import { Feature, StyleExpression } from "../../types";
+
+const JSONPATH_REGEX =
+  /\$[a-zA-Z_][a-zA-Z0-9_]*(\[[^\]]+\])?(\.[a-zA-Z_][a-zA-Z0-9_]*(\[[^\]]+\])?)*$/g;
```

Comment:
I think it's better to write a test about regex for JSONPath.
Because I tried to test this regex with `$.phoneNumbers[:1].type` in https://regex101.com/, but it doesn't work.